hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotUpdateSizeRequest.java

```diff
@@ -0,0 +1,98 @@
+import org.apache.hadoop.ozone.om.ratis.utils.OzoneManagerDoubleBufferHelper;
+import org.apache.hadoop.ozone.om.request.OMClientRequest;
+import org.apache.hadoop.ozone.om.request.util.OmResponseUtil;
+import org.apache.hadoop.ozone.om.response.OMClientResponse;
+import org.apache.hadoop.ozone.om.response.snapshot.OMSnapshotUpdateSizeResponse;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMRequest;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.SnapshotSize;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Updates the exclusive size of the snapshot.
+ */
+public class OMSnapshotUpdateSizeRequest extends OMClientRequest {
+
+  public OMSnapshotUpdateSizeRequest(OMRequest omRequest) {
+    super(omRequest);
+  }
+
+  @Override
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
+      long trxnLogIndex, OzoneManagerDoubleBufferHelper omDoubleBufferHelper) {
+
+    OMClientResponse omClientResponse = null;
+    OMMetadataManager metadataManager = ozoneManager.getMetadataManager();
+
+    OzoneManagerProtocolProtos.OMResponse.Builder omResponse =
+        OmResponseUtil.getOMResponseBuilder(getOmRequest());
+    OzoneManagerProtocolProtos.SnapshotUpdateSizeRequest
+        snapshotUpdateSizeRequest = getOmRequest()
+        .getSnapshotUpdateSizeRequest();
+
+    List<SnapshotSize> snapshotSizeList = snapshotUpdateSizeRequest
+        .getSnapshotSizeList();
+    Map<String, SnapshotInfo> updatedSnapInfos = new HashMap<>();
+
+    try {
+      for (SnapshotSize snapshotSize: snapshotSizeList) {
+        String snapshotKey = snapshotSize.getSnapshotKey();
+        long exclusiveSize = snapshotSize.getExclusiveSize();
+        long exclusiveReplicatedSize = snapshotSize
+            .getExclusiveReplicatedSize();
+        SnapshotInfo snapshotInfo = metadataManager
+            .getSnapshotInfoTable().get(snapshotKey);
+
+        if (snapshotInfo == null) {
+          continue;
+        }
+
+        // Set Exclusive size.
+        snapshotInfo.setExclusiveSize(exclusiveSize);
+        snapshotInfo.setExclusiveReplicatedSize(exclusiveReplicatedSize);
+        // Update Table Cache
+        metadataManager.getSnapshotInfoTable().addCacheEntry(
+            new CacheKey<>(snapshotKey),
+            CacheValue.get(trxnLogIndex, snapshotInfo));
+        updatedSnapInfos.put(snapshotKey, snapshotInfo);
+      }
+      omClientResponse = new OMSnapshotUpdateSizeResponse(
+          omResponse.build(), updatedSnapInfos);
+    } catch (IOException ex) {
+      omClientResponse = new OMSnapshotUpdateSizeResponse(
+          createErrorOMResponse(omResponse, ex));
+    } finally {
+      addResponseToDoubleBuffer(trxnLogIndex, omClientResponse,
+          omDoubleBufferHelper);
+    }
+    return omClientResponse;
```

Comment:
Do we need audit logging for this?

Reply:
I don't think we need to audit log. `try` is only used here for `snapshotInfoTable.get()`. So I'm not sure if we need to audit anything as we are only updating property. But I'm adding Logging `error` as mentioned in the above comment.

Reply:
Sounds good.